wa/workloads/aitutu/uiauto/app/src/main/java/com/arm/wa/uiauto/aitutu/UiAutomation.java

```diff
@@ -0,0 +1,147 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+
+package com.arm.wa.uiauto.aitutu;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.graphics.Rect;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiSelector;
+import android.support.test.uiautomator.UiScrollable;
+import android.view.KeyEvent;
+import android.util.Log;
+
+import com.arm.wa.uiauto.BaseUiAutomation;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeUnit;
+
+
+@RunWith(AndroidJUnit4.class)
+public class UiAutomation extends BaseUiAutomation {
+
+    public static String TAG = "UXPERF";
+
+    @Test
+    public void setup() throws Exception {
+        acceptPermissions();
+    }
+
+    @Test
+    public void runWorkload() throws Exception {
+        runBenchmark();
+    }
+
+    @Test
+    public void extractResults() throws Exception {
+        getScores();
+    }
+
+    public void acceptPermissions() throws Exception {
+        UiSelector selector = new UiSelector();
+        UiObject perm1 = mDevice.findObject(selector.textContains("1 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject perm2 = mDevice.findObject(selector.textContains("2 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject perm3 = mDevice.findObject(selector.textContains("3 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject perm4 = mDevice.findObject(selector.textContains("4 of 4")
+                                             .className("android.widget.TextView"));
+        UiObject allow = mDevice.findObject(selector.textContains("ALLOW")
+                                             .className("android.widget.Button"));
+        if (perm1.exists()) {
+            allow.click();
+        }
+        if (perm2.exists()) {
+            allow.click();
+        }
+        if (perm3.exists()) {
+            allow.click();
+        }
+        if (perm4.exists()) {
+            allow.click();
+        }
+
+        UiObject next = mDevice.findObject(selector.textContains("NEXT")
+                                             .className("android.widget.Button"));
+        waitObject(next);
```

Comment:
On my nexus5 I do not get any of these permission popups,  so the workload will wait here for 10 minutes before crashing. 
The `waitObject` calls should have a small timeout value and only attempt to select the object if they are present. 

Although are the permissions for the app not being automatically be granted during installation?

Reply:
I've removed this section as the permissions we're not visible when I ran through either. I added them when first creating the workload and running through manually.